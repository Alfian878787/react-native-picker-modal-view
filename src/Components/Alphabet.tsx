// Local Imports
import * as React from 'react';
import { View, Text } from 'react-native';

export class AlphabetComponent extends React.PureComponent {

	public render(): JSX.Element {
		return (
			<View>
				<Text>A</Text>
			</View>
		);
	}

}
