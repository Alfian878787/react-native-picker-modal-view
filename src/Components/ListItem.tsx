// Global Imports
import * as React from 'react';
import { Text, View, TouchableOpacity } from 'react-native';

// Local Import
import { IListItemInDto } from '@Interfaces';
import { ListItemStyle } from '@Styles';

export class ListItemComponent extends React.Component<IListItemInDto, {}> {
	public render(): JSX.Element {
		const { onSelectMethod } = this.props;
		const { Name } = this.props.list;
		return (
			<TouchableOpacity
				style={ListItemStyle.container}
				activeOpacity={0.7}
				onPress={() => onSelectMethod(this.props.list)}
			>
				<View style={ListItemStyle.btnContainer}>
					<Text>{Name}</Text>
				</View >
			</TouchableOpacity >
		);
	}
}
