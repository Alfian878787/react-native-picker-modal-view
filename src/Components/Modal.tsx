// Global Imports
import * as React from 'react';
import {
	Modal,
	View,
	FlatList,
	KeyboardAvoidingView,
	Dimensions,
	NativeSyntheticEvent,
	NativeScrollEvent,
} from 'react-native';

// Local Imports
import {
	AlphabetComponent,
	ListItemComponent,
	SearchComponent,
	ScrollToTopComponent,
	SelectBoxComponent,
} from '@Components';
import { IModalInDtoProps, IModalListInDto, IModalInDtoState } from '@Interfaces';
import { ModalStyles } from '@Styles';

const { height } = Dimensions.get('window');
export class ModalComponent extends React.Component<IModalInDtoProps, IModalInDtoState> {

	private flatListRef = null;
	private numToRender: number = 20;

	public state: IModalInDtoState = {
		modalVisible: false,
		searchText: '',
		stickyBottomButton: false,
		selectedAlpha: null,
		selectedObject: {} as IModalListInDto,
	};

	public static defaultProps = {
		animationType: 'slide',
		closeable: true,
		hideAlphabetFilter: false,
		placeholderTextColor: '#252525',
		autoGenerateAlphabet: false,
		sortingLanguage: 'tr',
		removeClippedSubviews: true,
		chooseText: 'Choose one...',
		searchText: 'Search anything...',
		autoCorrect: true,
		autoSort: false,
		list: [],
		disabled: false,
		forceSelect: false,
	};

	constructor(props: IModalInDtoProps) {
		super(props);
		this._onViewableItemsChanged = this._onViewableItemsChanged.bind(this);
	}

	public componentWillUnmount(): void {
		this.clearComponent();
	}

	public componentWillReceiveProps(nextProps, nextState): void {
		if (
			// tslint:disable-next-line: max-line-length
			(this.props.defaultSelected && nextProps.defaultSelected) && this.props.defaultSelected.Name !== nextProps.defaultSelected.Name &&
			this.props.defaultSelected.Id !== nextProps.defaultSelected.Id
		) {
			this.setState({
				selectedObject: {} as IModalListInDto,
			});
		}
	}

	private clearComponent(): void {
		this.setState({
			stickyBottomButton: false,
			searchText: '',
			selectedAlpha: null,
		});
	}

	public componentWillMount(): void {
		const { autoGenerateAlphabet, alphaBets } = this.props;
		if (autoGenerateAlphabet) {
			this.generateAlphabet();
		} else if (alphaBets) {
			this.setState({
				alphaBets,
			});
		}
	}

	private openModal(): void {
		const { list, autoGenerateAlphabet, disabled } = this.props;

		if (autoGenerateAlphabet) {
			this.generateAlphabet();
		}

		if (list.length > 0 && !disabled) {
			this.setState({
				modalVisible: true,
			});
		}
	}

	public render(): JSX.Element {
		const {
			animationType,
			onRequestClosed,
			closeable,
			hideAlphabetFilter,
			placeholderTextColor,
			keyExtractor,
			showToTopButton,
			onEndReached,
			removeClippedSubviews,
			FlatListProps,
			chooseText,
			searchText,
			autoCorrect,
			SearchInputProps,
			defaultSelected,
			disabled,
			list,
			forceSelect,
		} = this.props;
		const { modalVisible, alphaBets, stickyBottomButton, selectedAlpha, selectedObject } = this.state;
		return (
			<React.Fragment>
				<SelectBoxComponent
					disabled={(disabled || !list || list.length === 0)}
					selectedObject={selectedObject}
					chooseText={(defaultSelected && defaultSelected.Name) ? defaultSelected.Name : chooseText}
					openModal={this.openModal.bind(this)}
				/>
				<Modal
					animationType={animationType}
					visible={modalVisible}
					onRequestClose={() => onRequestClosed}>
					<View style={ModalStyles.container}>
						<SearchComponent
							autoCorrect={autoCorrect}
							searchText={searchText}
							placeholderTextColor={placeholderTextColor}
							onClose={this.onClose.bind(this)}
							onBackRequest={this.onBackRequest.bind(this)}
							closeable={!forceSelect && closeable}
							setText={(text: string) => this.setText(text)}
							{...SearchInputProps}
						/>
						<KeyboardAvoidingView style={ModalStyles.keyboardContainer} behavior="padding" enabled>
							<View style={ModalStyles.listArea}>

								<FlatList
									ref={(ref) => this.flatListRef = ref}
									data={this.getFilteredData()}
									keyExtractor={keyExtractor ? keyExtractor : this._keyExtractor.bind(this)}
									renderItem={({ item, index }) => this.renderItem(item, index)}
									onScroll={showToTopButton && this.onScrolling.bind(this)}
									initialNumToRender={this.numToRender}
									keyboardShouldPersistTaps={'always'}
									keyboardDismissMode={'on-drag'}
									onEndReached={onEndReached}
									removeClippedSubviews={removeClippedSubviews}
									viewabilityConfig={{
										minimumViewTime: 100,
										viewAreaCoveragePercentThreshold: 100,
										waitForInteraction: true,
									}}
									onViewableItemsChanged={this._onViewableItemsChanged}
									{...FlatListProps}
								/>

								{
									!hideAlphabetFilter &&
									<AlphabetComponent
										setAlphabet={(alphabet: string) => this.setAlphabet(alphabet)}
										alphaBets={alphaBets}
										selectedAlpha={selectedAlpha}
									/>
								}
							</View>
						</KeyboardAvoidingView>
						{stickyBottomButton && <ScrollToTopComponent goToUp={() => this.scrollToUp()} />}
					</View>
				</Modal>
			</React.Fragment >
		);
	}

	private _onViewableItemsChanged({ viewableItems, changed }): void {
		if (viewableItems && viewableItems[0]) {
			const firstLetter = viewableItems[0].item.Name.charAt(0);
			this.setState({
				selectedAlpha: firstLetter,
			});
		}
	}

	private onClose(): void {
		const { onRequestClosed, onSelected, forceSelect, defaultSelected } = this.props;
		const { modalVisible, selectedObject } = this.state;

		if (
			forceSelect &&
			(selectedObject && !selectedObject.Id) &&
			(defaultSelected && !defaultSelected.Id)
		) {
			return;
		}

		if (!forceSelect) {
			onSelected({} as IModalListInDto);
		}

		this.setState({
			selectedObject: {} as IModalListInDto,
			modalVisible: !modalVisible,
		});
		this.clearComponent();
		if (onRequestClosed) {
			onRequestClosed();
		}
	}

	private onBackRequest(): void {
		const { onBackRequest, onSelected } = this.props;
		const { modalVisible } = this.state;
		this.setState({
			modalVisible: !modalVisible,
		});
		this.clearComponent();
		if (onBackRequest) {
			onBackRequest();
		}
	}

	private scrollToUp(): void {
		if (this.flatListRef) {
			this.setState({
				selectedAlpha: null,
			}, () => {
				this.flatListRef.scrollToOffset({ animated: true, offset: 0 });
			});
		}
	}

	private onScrolling(e: NativeSyntheticEvent<NativeScrollEvent>): void {
		const { contentOffset } = e.nativeEvent;

		if (contentOffset.y > 100) {
			this.setState({
				stickyBottomButton: true,
			});
		} else {
			this.setState({
				stickyBottomButton: false,
			});
		}
	}

	private renderItem(item: IModalListInDto, index: number): JSX.Element {
		const { defaultSelected } = this.props;
		return <ListItemComponent
			defaultSelected={defaultSelected}
			list={item}
			onSelectMethod={this.onSelectMethod.bind(this)}
		/>;
	}

	private generateAlphabet(): void {
		const { list, sortingLanguage } = this.props;
		const singularAlpha = [];
		list.map((x: IModalListInDto) => {
			if (singularAlpha.indexOf(x.Name.charAt(0)) === -1) {
				singularAlpha.push(x.Name.charAt(0));
			}
		});

		singularAlpha.sort((a, b) => a.localeCompare(b, sortingLanguage));

		this.setState({
			alphaBets: singularAlpha,
		});
	}

	private _keyExtractor(item: IModalListInDto, index: number) {
		return index.toString();
	}

	private setText(text: string): void {
		this.setState({
			searchText: text,
		});
	}

	private compare(a: any, b: any): number {
		const aName = a.Name.toLocaleLowerCase();
		const bName = b.Name.toLocaleLowerCase();

		let comparison = 0;
		if (aName > bName) {
			comparison = 1;
		} else if (aName < bName) {
			comparison = -1;
		}
		return comparison;
	}

	private getFilteredData(): IModalListInDto[] {
		const { list, autoSort } = this.props;
		const { searchText } = this.state;

		if (autoSort) {
			list.sort(this.compare);
		}
		return list.filter((l: IModalListInDto) => l.Name.toLocaleLowerCase().indexOf(searchText.toLocaleLowerCase()) > -1);
	}

	private onSelectMethod(key: IModalListInDto): IModalListInDto | void {
		const { onSelected } = this.props;
		this.setState({
			modalVisible: false,
			selectedObject: key as IModalListInDto,
		});
		this.clearComponent();

		if (key && !key.Id) {
			return onSelected({} as IModalListInDto)
		}

		return onSelected(key);
	}

	private getIndex(alphabet: string): number {
		const list = this.getFilteredData();
		const findIndex = list.findIndex((x: IModalListInDto) => {
			return x.Name.charAt(0) === alphabet;
		});
		return findIndex;
	}

	private setAlphabet(alphabet: string): void {
		// const { list } = this.props;
		this.setState({
			selectedAlpha: alphabet,
		}, () => {
			const list = this.getFilteredData();
			const findIndex = this.getIndex(alphabet);
			if (findIndex >= 0 && findIndex <= (list.length - (this.numToRender / 2))) {
				this.flatListRef.scrollToIndex({ animated: true, index: findIndex, viewPosition: 0 });
			} else {
				this.flatListRef.scrollToEnd();
			}
		});
	}
}
