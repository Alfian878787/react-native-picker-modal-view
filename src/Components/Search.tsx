// Global Imports
import * as React from 'react';
import { View, TextInput, TouchableOpacity, Image } from 'react-native';

// Local Imports
import { SearchStyle } from '@Styles';
import { ISearch } from '@Interfaces';

export class SearchComponent extends React.PureComponent<ISearch, {}> {
	public render(): JSX.Element {
		const {
			SearchInputProps,
			autoCorrect,
			placeholderTextColor,
			onClose,
			setText,
			closeable,
			searchText,
			onBackRequest,
		} = this.props;
		return (
			<View style={SearchStyle.searchArea}>

				<TouchableOpacity onPress={() => onBackRequest()}>
					<Image source={require('../Assets/Images/left-arrow.png')} style={SearchStyle.backButton} />
				</TouchableOpacity>

				<TextInput
					placeholder={searchText}
					placeholderTextColor={placeholderTextColor}
					style={SearchStyle.textInput}
					underlineColorAndroid={'transparent'}
					onChangeText={(text: string) => setText(text)}
					autoCorrect={autoCorrect}
					{...SearchInputProps}
				/>
				{closeable &&
					<TouchableOpacity onPress={() => onClose()}>
						<Image source={require('../Assets/Images/close.png')} style={SearchStyle.closeButton} />
					</TouchableOpacity>
				}
			</View>
		);
	}
}
