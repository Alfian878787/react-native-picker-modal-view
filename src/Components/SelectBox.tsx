// Global Imports
import * as React from 'react';
import { View, Text, TouchableOpacity, Image } from 'react-native';

// Local Imports
import { ISelectBoxProps } from '@Interfaces';
import { SelectBoxStyle } from '@Styles';

export class SelectBoxComponent extends React.PureComponent<ISelectBoxProps, {}> {
	public render(): JSX.Element {
		const { openModal, selectedObject, chooseText } = this.props;
		return (
			<TouchableOpacity
				activeOpacity={0.7}
				onPress={() => openModal()}
			>
				<View style={SelectBoxStyle.container}>
					{
						!selectedObject ?
							<Text style={SelectBoxStyle.chooseText}>{chooseText}</Text> :
							<Text style={SelectBoxStyle.chooseText}>{selectedObject.Name}</Text>
					}
					<Image source={require('../Assets/Images/down.png')} style={SelectBoxStyle.downBtn} />
				</View>
			</TouchableOpacity>
		);
	}
}
