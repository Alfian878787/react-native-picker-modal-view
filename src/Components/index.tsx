export { ModalComponent } from '@Components/Modal';
export { AlphabetComponent } from '@Components/Alphabet';
export { ListItemComponent } from '@Components/ListItem';
export { SearchComponent } from '@Components/SearchComponent';
export { ScrollToTopComponent } from '@Components/ScrollToTopComponent';
