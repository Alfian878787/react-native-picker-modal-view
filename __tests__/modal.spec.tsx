import * as React from 'react';
import { Button } from 'react-native';
import { shallow, ShallowWrapper, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import { ModalComponent } from '../src/Components/Modal';
import { any } from 'prop-types';

describe('ModalComponent', () => {
	describe('rendering', () => {
		let wrapper: ShallowWrapper;
		let mounting;
		const data = {
			Id: 1,
			Name: 'John Doe',
			Value: 'John Doe',
		};

		const list = [
			{
				Id: 1,
				Name: 'John Doe',
				Value: 'John Doe',
			},
			{
				Id: 2,
				Name: 'Ali Doe',
				Value: 'Ali Doe',
			},
			{
				Id: 3,
				Name: 'Çetin Doe',
				Value: 'Çetin Doe',
			},
			{
				Id: 4,
				Name: 'Şule Doe',
				Value: 'Şule Doe',
			}
		];

		const onSelected = jest.fn((a) => a);
		const onRequestClosed = jest.fn();
		const onBackRequest = jest.fn();
		const onEndReached = jest.fn();
		const state = {
			modalVisible: false,
			searchText: '',
			stickyBottomButton: false,
			selectedAlpha: null,
			selectedObject: {},
		}

		const component = (<ModalComponent
			showAlphabeticalIndex={false}
			onClosed={onRequestClosed}
			onBackButtonPressed={onBackRequest}
			onSelected={onSelected}
			items={list}
			alphabeticalIndexChars={['A', 'B', 'C', 'D', 'E']}
			searchInputTextColor={'#ddd'}
			autoGenerateAlphabeticalIndex={true}
			showToTopButton={true}
			onEndReached={onEndReached}
			removeClippedSubviews={false}
			selectPlaceholderText={'Choose one...'}
			selected={data}
			searchPlaceholderText={'Search...'}
			autoSort={false}
			disabled={false}
			requireSelection={false}
		/>);

		beforeEach(() => {
			wrapper = shallow(component);
			mounting = mount(component);
		});

		test('should render a Modal Component', () => {
			const render = renderer.create(component).toJSON();
			expect(typeof wrapper.state('modalVisible')).toBe('boolean');
			expect(typeof wrapper.state('searchText')).toBe('string');
			expect(wrapper.state('selectedAlpha')).toBeDefined;
			expect(typeof wrapper.state('stickyBottomButton')).toBe('boolean');
			expect(typeof wrapper.state('selectedObject')).toBe('object');
			expect(render).toMatchSnapshot();
		});


		test('should (showAlphabeticalIndex,autoGenerateAlphabeticalIndex,autoCorrect,autoSort,disabledforceSelect) type is boolean ', () => {
			expect(typeof mounting.props().showAlphabeticalIndex).toBe('boolean');
			expect(typeof mounting.props().autoGenerateAlphabeticalIndex).toBe('boolean');
			expect(typeof mounting.props().autoSort).toBe('boolean');
			expect(typeof mounting.props().requireSelection).toBe('boolean');
		});

		test('should fired onSelected prop', () => {
			expect(typeof mounting.props().onSelected).toBe('function');
		});

		test('should fired onClosed prop', () => {
			expect(typeof mounting.props().onClosed).toBe('function');
		});

		test('should fired onBackButtonPressed prop', () => {
			expect(typeof mounting.props().onBackButtonPressed).toBe('function');
		});

		test('should fired onEndReached prop', () => {
			expect(typeof mounting.props().onEndReached).toBe('function');
		});

		test('should clearComponent function set default state', () => {
			const spy = spyOn(ModalComponent.prototype, 'clearComponent').and.callThrough();
			const instance = mounting.instance() as any;
			instance.clearComponent();
			expect(spy).toHaveBeenCalledTimes(1);
			expect(wrapper.state()).toMatchObject({
				stickyBottomButton: false,
				searchText: '',
				selectedAlpha: null
			});
		});

		test('should openmodal fired modal is open', () => {
			const spy = spyOn(ModalComponent.prototype, 'openModal').and.callThrough();
			const instance = mounting.instance() as any;
			mounting.setProps({
				disabled: false
			});
			instance.openModal();
			expect(spy).toHaveBeenCalledTimes(1);
			setTimeout(() => {
				expect(wrapper.state('modalVisible')).toBeTruthy();
			}, 0);
		});

		test('should onclose fired modal close', () => {
			const spy = spyOn(ModalComponent.prototype, 'onClose').and.callThrough();
			const instance = mounting.instance() as any;
			instance.onClose();
			expect(spy).toHaveBeenCalledTimes(1);
			setTimeout(() => {
				expect(wrapper.state()).toMatchObject({
					selectedObject: {},
					modalVisible: false
				})
			}, 0);
		});

		test('should onbackrequest fired back', () => {
			const spy = spyOn(ModalComponent.prototype, 'onBackRequest').and.callThrough();
			const instance = mounting.instance() as any;
			instance.onBackRequest();
			expect(spy).toHaveBeenCalledTimes(1);
			setTimeout(() => {
				expect(wrapper.state('modalVisible')).not.toBeTruthy();
			}, 0);
		});

		test('should scrollToUp fired flatlist scroll to offset 0 and selectedAlpha set to null', () => {
			const spy = spyOn(ModalComponent.prototype, 'scrollToUp').and.callThrough();
			const instance = mounting.instance() as any;
			instance.scrollToUp();
			expect(spy).toHaveBeenCalledTimes(1);
			setTimeout(() => {
				expect(wrapper.state('selectedAlpha')).not.toBeTruthy();
			}, 0);
		});

		test('should auto generate alphabets array with list', () => {
			const spy = spyOn(ModalComponent.prototype, 'generateAlphabet').and.callThrough();
			const instance = mounting.instance() as any;
			mounting.setProps({
				list
			});

			const stateLength: string[] = mounting.state('alphabeticalIndexChars').length;
			const propsLength: string[] = mounting.props().list.length;

			expect(stateLength).toEqual(propsLength);

			instance.generateAlphabet();
			expect(spy).toHaveBeenCalledTimes(1);
		});

		test('should text set to state', () => {
			const spy = spyOn(ModalComponent.prototype, 'setText').and.callThrough();
			const instance = mounting.instance() as any;
			const text = 'Test Text';

			wrapper.setState({
				searchText: text
			});

			expect(wrapper.state('searchText')).toEqual(text);

			instance.setText(text);
			expect(spy).toHaveBeenCalled();
		});

		test('should get filtered list data', () => {
			const spy = spyOn(ModalComponent.prototype, 'getFilteredData').and.callThrough();
			const instance = wrapper.instance() as any;

			// get all list
			expect(instance.getFilteredData()).toMatchObject(list);

			wrapper.setState({
				searchText: 'Şule'
			});

			expect(spy).toHaveBeenCalled();

			// get filtered list
			expect(instance.getFilteredData()).toMatchObject([{
				Id: 4,
				Name: 'Şule Doe',
				Value: 'Şule Doe',
			}]);
		});

		test('should get selected', () => {
			const spy = spyOn(ModalComponent.prototype, 'onSelectMethod').and.callThrough();
			const instance = wrapper.instance() as any;
			const dummyData = {
				Id: 4,
				Name: 'Şule Doe',
				Value: 'Şule Doe',
			};

			wrapper.setState(dummyData);

			expect(instance.onSelectMethod(dummyData)).toMatchObject(dummyData);
			expect(wrapper.state()).toMatchObject({
				modalVisible: false,
				selectedObject: dummyData
			});
			expect(instance.onSelectMethod()).toBeUndefined();
			expect(spy).toBeCalled()
		});

		test('should get index in list array', () => {
			const spy = spyOn(ModalComponent.prototype, 'getIndex').and.callThrough();
			const instance = mounting.instance() as any;

			// A defined in the prefined list
			expect(instance.getIndex('A')).toEqual(1);

			// Z not defined in the predefined list
			expect(instance.getIndex('Z')).toEqual(-1);
			expect(spy).toHaveBeenCalled();
		});

		test('should press and set alphabet, scroll offset for flatlist', () => {
			const spy = spyOn(ModalComponent.prototype, 'setAlphabet').and.callThrough();
			const instance = mounting.instance() as any;
			const selectedAlpha = 'A';

			wrapper.setState({
				selectedAlpha
			});

			instance.setAlphabet(selectedAlpha);
			expect(spy).toHaveBeenCalled();

			setTimeout(() => {
				expect(wrapper.state('selectedAlpha')).toEqual(selectedAlpha);
			}, 0);
		});

		test('should render a renderListItem', () => {
			const selectListItemView = () => <Button title="renderListItem" onPress={jest.fn()} />;
			mounting.setProps({
				renderListItem: selectListItemView
			});

			expect(mounting.find({ title: "renderListItem" })).toHaveLength(list.length);
		});

		test('should render a renderSelectView', () => {
			const selectView = () => <Button title="renderSelectView" onPress={jest.fn()} />;
			mounting.setProps({
				renderSelectView: selectView
			});

			expect(mounting.find({ title: "renderSelectView" })).toHaveLength(1);
		});

	});
});
