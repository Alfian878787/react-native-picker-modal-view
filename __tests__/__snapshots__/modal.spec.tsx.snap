// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalComponent rendering should render a Modal Component 1`] = `
Array [
  <react-native-mock>
    <react-native-mock>
      <react-native-mock>
        John Doe
      </react-native-mock>
      <react-native-mock />
    </react-native-mock>
  </react-native-mock>,
  <react-native-mock>
    <react-native-mock>
      <react-native-mock>
        <react-native-mock>
          <react-native-mock />
        </react-native-mock>
        <react-native-mock />
        <react-native-mock>
          <react-native-mock />
        </react-native-mock>
      </react-native-mock>
      <react-native-mock>
        <react-native-mock>
          <react-native-mock>
            <react-native-mock>
              <react-native-mock>
                <react-native-mock>
                  John Doe
                </react-native-mock>
              </react-native-mock>
            </react-native-mock>
            <react-native-mock>
              <react-native-mock>
                <react-native-mock>
                  Ali Doe
                </react-native-mock>
              </react-native-mock>
            </react-native-mock>
            <react-native-mock>
              <react-native-mock>
                <react-native-mock>
                  Çetin Doe
                </react-native-mock>
              </react-native-mock>
            </react-native-mock>
            <react-native-mock>
              <react-native-mock>
                <react-native-mock>
                  Şule Doe
                </react-native-mock>
              </react-native-mock>
            </react-native-mock>
          </react-native-mock>
          <react-native-mock>
            <react-native-mock>
              <react-native-mock>
                <react-native-mock>
                  A
                </react-native-mock>
              </react-native-mock>
              <react-native-mock>
                <react-native-mock>
                  Ç
                </react-native-mock>
              </react-native-mock>
              <react-native-mock>
                <react-native-mock>
                  J
                </react-native-mock>
              </react-native-mock>
              <react-native-mock>
                <react-native-mock>
                  Ş
                </react-native-mock>
              </react-native-mock>
            </react-native-mock>
          </react-native-mock>
        </react-native-mock>
      </react-native-mock>
    </react-native-mock>
  </react-native-mock>,
]
`;
